client/events.go

```diff
@@ -1,20 +1,70 @@
 package client
 
 import (
-	"io"
+	"encoding/json"
 	"net/url"
 	"time"
 
 	"golang.org/x/net/context"
 
 	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/events"
 	"github.com/docker/engine-api/types/filters"
 	timetypes "github.com/docker/engine-api/types/time"
 )
 
-// Events returns a stream of events in the daemon in a ReadCloser.
-// It's up to the caller to close the stream.
-func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (io.ReadCloser, error) {
+// Events returns a stream of events in the daemon. It's up to the caller to close the stream
+// by cancelling the context. Once the stream has been completely read an io.EOF error will
+// be sent over the error channel. If an error is sent all processing will be stopped. It's up
+// to the caller to reopen the stream in the event of an error by reinvoking this method.
+func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (<-chan events.Message, <-chan error) {
+
+	messages := make(chan events.Message, 1)
+	errs := make(chan error, 1)
```

Comment:
@stevvooe You mentioned in previous comment to make this a buffered channel in case the error wasn't consumed, but when you do this you force the consumer to check the event channel after an error has been received because both the event and error channel can have data. 

Ex)
```
for {
  evt := <- eventq:
     // Do something
  err := <- errq:
     return err
     // it's possible you just missed the last event
}
```

```
for {
  evt := <- eventq:
     // Do something
  err := <- errq:
     // boo error
     select {
          case evt := <- eventq:
               // make sure no left over events
          default:
               // guess no events
     }
     return err
}
```

That seems not great for someone using this method since it's not clear you need to do this w/o looking at the source code.